Gems/Multiplayer/Code/Source/Components/NetworkDebugPlayerIdComponent.cpp

```diff
@@ -0,0 +1,185 @@
+            {
+                editContext->Class<NetworkDebugPlayerIdComponent>("Network Debug Connection Counter", "Renders the player id as debug text over network players.")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+                        ->Attribute(AZ::Edit::Attributes::Category, "Multiplayer")
+                        ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &NetworkDebugPlayerIdComponent::m_translationOffset, "Translation Offset", "The world-space offset from the player position to render the debug text.")
+                    ->DataElement(AZ::Edit::UIHandlers::Default, &NetworkDebugPlayerIdComponent::m_fontScale, "Font Scale", "Apply a scale to the default debug font rendering size.")
+                    ->DataElement(AZ::Edit::UIHandlers::Color, &NetworkDebugPlayerIdComponent::m_fontColor, "Color", "Debug text color.")
+                ;
+            }
+        }
+
+        NetworkDebugPlayerIdComponentBase::Reflect(context);
+    }
+
+    void NetworkDebugPlayerIdComponent::OnActivate([[maybe_unused]] EntityIsMigrating entityIsMigrating)
+    {
+        #if (!AZ_TRAIT_CLIENT)
+            return;
+        #endif
+
+        m_viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
+        if (!m_viewport)
+        {
+            AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to find the a rendering viewport. Debug rendering will be disabled.");
+            return;
+        }
+
+        const auto fontQueryInterface = AZ::Interface<AzFramework::FontQueryInterface>::Get();
+        if (!fontQueryInterface)
+        {
+            AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to find the FontQueryInterface. Debug rendering will be disabled.");
+            return;
+        }
+
+        m_fontDrawInterface = fontQueryInterface->GetDefaultFontDrawInterface();
+        if (!m_fontDrawInterface)
+        {
+            AZ_Assert(false, "NetworkDebugPlayerIdComponent failed to find the FontDrawInterface. Debug rendering will be disabled.");
+            return;
+        }
+
+        m_drawParams.m_drawViewportId = m_viewport->GetId();
+        m_drawParams.m_scale = AZ::Vector2(m_fontScale);
+        m_drawParams.m_color = m_fontColor;
+
+        AZ::TickBus::Handler::BusConnect();
+    }
+
+    void NetworkDebugPlayerIdComponent::OnDeactivate([[maybe_unused]] EntityIsMigrating entityIsMigrating)
+    {
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void NetworkDebugPlayerIdComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+    {
+        if (!cl_debugPlayerIdRender)
+        {
+            return;
+        }
+
+        AZ::Vector3 renderWorldSpace = GetEntity()->GetTransform()->GetWorldTranslation() + m_translationOffset;
```

Comment:
You could move this after the if checks that bail out.

Reply:
The if-check uses the world space position (making sure we dont render players behind the player)